meterfs/crypt.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Meterfs cryptography
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+
+#include "tinyaes/aes.h"
+#include "crypt.h"
+
+
+static unsigned int meterfs_hash(unsigned int seed)
+{
+	int result;
+
+	seed *= 1103515245;
+	seed += 12345;
+	result = (seed >> 16) & 0x7ff;
+
+	seed *= 1103515245;
+	seed += 12345;
+	result <<= 10;
+	result ^= (seed >> 16) & 0x3ff;
+
+	seed *= 1103515245;
+	seed += 12345;
+	result <<= 10;
+	result ^= (seed >> 16) & 0x3ff;
+
+	return result;
+}
+
+
+void meterfs_constructIV(uint8_t *iv, const file_t *f, const entry_t *e)
+{
+	static const uint8_t filler[16] = {
+		0x55, 0xaa, 0x55, 0xaa, 0x55, 0xaa, 0x55, 0xaa,
+		0x55, 0xaa, 0x55, 0xaa, 0x55, 0xaa, 0x55, 0xaa
+	};
+	unsigned liv[4] = {
```

Comment:
This will produce different results based on int size and endianness.

Reply:
Right now meterfs is host-endian anyway, so I'm not sure it matters

Reply:
Added serialization